finality-aleph/src/sync/ticker.rs

```diff
@@ -0,0 +1,143 @@
+use tokio::time::{sleep, Duration, Instant};
+
+/// This struct is used for determining when we should broadcast justification so that it does not happen too often.
+pub struct BroadcastTicker {
+    last_broadcast: Instant,
+    current_periodic_timeout: Duration,
+    max_timeout: Duration,
+    min_timeout: Duration,
+}
+
+impl BroadcastTicker {
+    pub fn new(max_timeout: Duration, min_timeout: Duration) -> Self {
+        Self {
+            last_broadcast: Instant::now(),
+            current_periodic_timeout: max_timeout,
+            max_timeout,
+            min_timeout,
+        }
+    }
+
+    /// Returns whether we should broadcast right now if we just imported a justification.
```

Comment:
Just to be clearer about the lengthy comment above – e.g. this could be called `try_tick` and it would return whether the tick was successful.

The trigger to me writing these comments was the imo too detailed explanation of batches and justifications, generically, we just don't want the ticks to happen too often, and it's our private matter as to why. :P The comment about (potentially) broadcasting two justifications per batch likely should end up somewhere where this will be used rather than in this documentation.

Reply:
Changed after renaming to `Ticker`